src/scalar/binary/scalar.rs

```diff
@@ -58,7 +70,10 @@ impl<O: OffsetSizeTrait> From<WKB<'_, O>> for geo::Geometry {
 impl<O: OffsetSizeTrait> From<&WKB<'_, O>> for geo::Geometry {
     fn from(value: &WKB<'_, O>) -> Self {
         let buf = value.arr.value(value.geom_index);
-        geozero::wkb::Wkb(buf.to_vec()).to_geo().unwrap()
+        match value.flavor {
+            WKBFlavor::ISO => geozero::wkb::Wkb(buf.to_vec()).to_geo().unwrap(),
+            WKBFlavor::EWKB => geozero::wkb::Ewkb(buf.to_vec()).to_geo().unwrap(),
```

Comment:
This is a very small subset of the wkb parsing in this crate. The majority happens when casting a wkb array to a concretely-typed geometry array. That happens with our own WKB parsing (see `crate::io::wkb` iirc). As a precursor to ewkb support, we need to support parsing ewkb geometries with those functions, or else ewkb will still crash on the vast majority of operations

Reply:
This exceed my knowledge. I don't know the binary format difference between wkb flavors yet. Now I mainly use WKBArray instead of concretely-typed array. Maybe we can merge this pr but still keep the issue open? I will learn them, it'll nice if you have time to create pr to fully support ewkb.